index.js

```diff
@@ -458,15 +459,42 @@ class ClientServiceBase {
   /**
    * Execute a registered contract
    * @param {string} contractId
-   * @param {Object} argument
-   * @param {Object} [functionArgument=undefined]
+   * @param {Object|string} contractArgument
+   * @param {Object|string} [functionArgument=null]
+   * @param {string} [functionId=""]
+   * @param {string} [nonce=null]
    * @return {Promise<ContractExecutionResult|void|*>}
    * @throws {ClientError|Error}
    */
-  async executeContract(contractId, argument, functionArgument) {
+  async executeContract(
+      contractId,
+      contractArgument,
+      functionArgument = null,
+      functionId = '',
+      nonce = null,
```

Comment:
This is the main change.

The Java client SDK uses function overloading to support backward compatibility.
Unfortunately, JavaScript doesn't provide function overloading.

My original thinking is to have another function called `executeContractAndFunction`, however, after Yamada-san accepts that we only use V2 contract argument formatting in the Node client SDK, I decided to withdraw the `executeContractAndFunction` plan.

A bit disadvantage is that the order of the function arguments looks a bit weird.

Reply:
@supl Thank you for the update. Is there any reason for using `''` instead of `null` for the default value of `functionId`. I guess you discussed it with Hiro in another PR, but I don't remember. Sorry.

Reply:
Sorry, I'm not sure, but why does the order of the arguments need to be like that?

Reply:
@jnmt 
Thank you for your question.
I guess it was because I thought the function_id field of the gRPC message was a string. So if I set the default value to an empty string, I could save a defensive statement to check if function_id is not passed by the user. But, I just realized because I have this defensive statement below anyway for the array

```
const functionIds =
   typeof functionId === 'string' && functionId.length > 0 ?
     [functionId] :
     [];
``` 
Therefore, I think it's safe to use null as the default value here for better consistency.

Thank you for your reminder.
I fixed this in https://github.com/scalar-labs/scalardl-javascript-sdk-base/pull/83/commits/4dd995a05ed1f61cd5553916fb8e954f12668e61

Please take a look again.
Thank you!

Reply:
> Sorry, I'm not sure, but why does the order of the arguments need to be like that?

@feeblefakie 
Thank you for the question.
This is because the original function interface is
```
async executeContract(contract_id, contract_argument, function_argument)
```

and I want to keep using only one function for executing contracts and also try to keep it backward compatible with the existing applications. So, I append the additional arguments.

Reply:
Sorry, I'm still not sure about what you mean.
So, what is exactly the problem when defining like below? (I mainly talk about the order of arguments)

async executeContract(
      contractId,
      contractArgument,
      functionId,
      functionArgument,
      nonce);

I think, as long as you set protobuf messages properly (in a backward-compatible way),
you can just define a new method with arbitrary arguments order.
The new method has nothing to do with the old `executeContract` method defined in 3.4, right?

Reply:
@feeblefakie 

Sorry for my lack of explanation.

> So, what is exactly the problem when defining like below? (I mainly talk about the order of arguments)

If I change current 

`executeContract(contractId, contractArgument, functionArgument)` to `executeContractAndFunction(contractId, contractArgument, functionId, functionArgument, nonce)`, then we need to rewrite the applications.

(Although I think maybe it's also fine because we don't really have any Node.js middleware applications. What we have now are the applications for testing purposes)

I can not define another function with the name `executeContract` because JavaScript doesn't support the function overloading.

My original thinking was to have two functions

```
executeContract(contractId, contractArgument, functionArgument) // use v1 serialization
```

and

```
executeContractAndFunction(contractId, contractArgument, functionId, functionArgument, nonce) // use v2 serialization
```

Reply:
Ah, OK, I got it. Thank you for the explanation.
I think the original idea sounds better, but not fully sure.
How about naming it `execute`?
Also, in Javascript, isn't it more common to define an argument as a map, so the ordering of arguments doesn't matter?

Reply:
@feeblefakie 
Thanks for your feedback.

Understood.
Then, I preserved the original interface `async executeContract(contractId, contractArgument, functionArgument)` and created another `async execute(contractId, contractArgument, functionId, functionArgument, nonce)` in https://github.com/scalar-labs/scalardl-javascript-sdk-base/pull/83/commits/9e2119ba781b4c1afca99fa110df5f830da3b889

Please take a look.
Thank you!

> Also, in Javascript, isn't it more common to define an argument as a map, so the ordering of arguments doesn't matter? (I'm just not sure)
It's not a common practice from my understanding.
I guess the reason is it's hard to know the interface from just reading the function.